SkimsAUX/plugins/PDFWeightProducer.cc

```diff
@@ -55,7 +62,10 @@ namespace LHAPDF {
 PDFWeightProducer::PDFWeightProducer(const edm::ParameterSet& iConfig) : getterOfProducts_(edm::ProcessMatch("*"), this)
 {
    callWhenNewProductsRegistered(getterOfProducts_);
-      
+   //GenSrc_== iConfig.getParameter<GenEventInfoProduct>("generatorSource");
+   //tracks_ = consumes<reco::TrackCollection>(iConfig.getUntrackedParameter<edm::InputTag>("src",edm::InputTag("generalTracks")));
+   GenTok_ = consumes<GenEventInfoProduct>(iConfig.getUntrackedParameter<edm::InputTag>("src",edm::InputTag("generator")));
+     
```

Comment:
I haven't tested this, but do you have to use getUntrackedParameter?
Does this work?
GenTok_ = consumes<GenEventInfoProduct>(edm::InputTag("generator"))